model/src/main/kotlin/Identifier.kt

```diff
@@ -68,11 +68,11 @@ data class Identifier(
             .thenComparing({ it.version }, AlphaNumericComparator)
     }
 
-    private constructor(components: List<String>) : this(
-        type = components.getOrElse(0) { "" },
-        namespace = components.getOrElse(1) { "" },
-        name = components.getOrElse(2) { "" },
-        version = components.getOrElse(3) { "" }
+    private constructor(properties: List<String>) : this(
```

Comment:
Maybe `coordinates` could be a better fit?

Reply:
I'd prefer to keep `properties` (even if a bit generic) as it aligns with the docs for `EMPTY` and `COMPARATOR`, and also the `toCoordinates()` function name is probably less confusing then. ("Why would you convert something to coordinates that already are coordinates?")